src/redux/books/booksSlice.js

```diff
@@ -42,8 +65,46 @@ const booksSlice = createSlice({
       return { ...state, books: updatedBooks };
     },
   },
+  extraReducers: {
+    [getBooks.pending]: (state) => ({ ...state, isLoading: true }),
+    [getBooks.fulfilled]: (state, action) => {
+      const newBooks = [];
+      const gotBooks = action.payload;
+      Object.keys(action.payload).forEach((book) => newBooks.push({
+        item_id: book,
+        title: gotBooks[book][0].title,
+        author: gotBooks[book][0].author,
+        category: gotBooks[book][0].category,
+      }));
+      return {
+        ...state,
+        isLoading: false,
+        books: newBooks,
+      };
+    },
+    [addBook.rejected]: (state, action) => ({
+      ...state,
+      error: action.payload,
+    }),
+    [addBook.pending]: (state) => ({ ...state, isLoading: true, error: false }),
+    [addBook.fulfilled]: (state) => ({
+      ...state,
+      isLoading: false,
+      error: false,
+    }),
+    [addBook.rejected]: (state) => ({ ...state, error: true }),
+    [removeBook.pending]: (state) => ({
+      ...state,
+      isLoading: true,
+      error: false,
+    }),
+    [removeBook.fulfilled]: (state, { payload }) => {
+      const id = payload;
+      const bookSliceState = state;
+      bookSliceState.books = state.books.filter((book) => book.item_id !== id);
+      bookSliceState.error = false;
+    },
```

Comment:
- [ ] Adding and Removing books is getting persisted fine. But I have to reload the page every time I add or remove a book to see the changes. KIndly dispatch actions to update the local state so the UI is updated automatically. ✍️